src/writer.rs

```diff
@@ -333,6 +333,12 @@ impl FdtWriter {
             .try_into()
             .map_err(|_| Error::TotalSizeTooLarge)?;
 
+        // The following operation cannot fail because we previously checked
+        // that both `data.len()` and `strings.len()` fit in a u32, thus
+        // the addition fits in an usize.
```

Comment:
Is this still valid on 32bit platforms? Can those values be somehow close to u32::MAX?

Reply:
Good catch. I'll use checked add here.